lib/collection/src/shard/collection_shard_distribution.rs

```diff
@@ -1,55 +1,66 @@
 use std::collections::HashMap;
 
+use crate::collection_state::ShardInfo;
 use crate::shard::{PeerId, ShardId};
 
+#[derive(Debug)]
+pub enum ShardType {
+    Local,
+    Remote(PeerId),
+    ReplicaSet { local: bool, remote: Vec<PeerId> },
```

Comment:
I have a vague memory of a conversation but I'd rather ask again :)

Do we enforce that there cannot be more than 1 replica per peer for a given shard?

Reply:
Yes, that's the plan. I can add a HashSet here if that's what you mean :smile: